java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -143,6 +151,19 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
       return;
     }
     switch (instruction.opcode) {
+      case ICONST_0:
+        SymbolicValue svZero = constraintManager.createSymbolicValue(instruction);
+        programState = programState.stackValue(svZero).addConstraint(svZero, DivisionByZeroCheck.ZeroConstraint.ZERO);
```

Comment:
BooleanConstraint.FALSE should also be added, to have correct behavior for boolean literals.